sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -138,9 +140,29 @@ public String getPath() {
      * @param queryParameterName The name of the query parameter.
      * @param queryParameterEncodedValue The encoded value of the query parameter.
      * @return The provided query parameter name and encoded value to query string for the final URL.
+     * @throws NullPointerException if {@code queryParameterName} or {@code queryParameterEncodedValue} are null.
      */
     public UrlBuilder setQueryParameter(String queryParameterName, String queryParameterEncodedValue) {
-        query.put(queryParameterName, queryParameterEncodedValue);
+        query.put(queryParameterName, new QueryParameter(queryParameterName, queryParameterEncodedValue));
+        return this;
+    }
+
+    /**
+     * Append the provided query parameter name and encoded value to query string for the final URL.
+     *
+     * @param queryParameterName The name of the query parameter.
+     * @param queryParameterEncodedValue The encoded value of the query parameter.
+     * @return The provided query parameter name and encoded value to query string for the final URL.
+     * @throws NullPointerException if {@code queryParameterName} or {@code queryParameterEncodedValue} are null.
+     */
+    public UrlBuilder appendQueryParameter(String queryParameterName, String queryParameterEncodedValue) {
```

Comment:
I'm not sure we should call this `append`, I think `add` is probably appropriate.

Reply:
Done.